snap/types.go

```diff
@@ -139,10 +139,13 @@ func (confinementType *ConfinementType) fromString(str string) error {
 type ServiceStopReason string
 
 const (
-	StopReasonRefresh ServiceStopReason = "refresh"
-	StopReasonRemove  ServiceStopReason = "remove"
-	StopReasonDisable ServiceStopReason = "disable"
-	StopReasonOther   ServiceStopReason = ""
+	StopReasonRefresh            ServiceStopReason = "refresh"
+	StopReasonRemove             ServiceStopReason = "remove"
+	StopReasonDisable            ServiceStopReason = "disable"
+	StopReasonQuotaGroupModified ServiceStopReason = "quota-group-modified"
```

Comment:
Nice!

Reply:
the additions are not in use anymore afaict

Reply:
removed, though I left some food for thought in the SU about our current pattern of restarting services like we do in this PR - I copied what we do for `snap restart` here, but I think that's wrong since it introduces race conditions for some types of services that are very chatty and don't handle the other end disappearing very well.

I think we should land this PR as-is and then followup with a refactor, possibly also after https://github.com/snapcore/snapd/pull/10266